plugins/workbench/src/index.ts

```diff
@@ -79,7 +79,7 @@ export interface SpecialNavModel {
   componentProps?: Record<string, string>
   // If not top and bottom, position will be sorted alphabetically.
   position?: 'top' | 'bottom' | string // undefined == 'top
-  visibleIf?: Resource<(spaces: Space[]) => boolean>
+  visibleIf?: Resource<(spaces: Space[]) => Promise<boolean>>
```

Comment:
You need to update all usages of visibleIf.

Reply:
Done